crates/zizmor/src/finding/location.rs

```diff
@@ -96,6 +96,20 @@ impl<'doc> From<Vec<RouteComponent<'doc>>> for Route<'doc> {
     }
 }
 
+impl<'doc> Route<'doc> {
+    /// Convert this location Route to a yamlpatch Route
+    pub(crate) fn to_yamlpatch_route(&self) -> yamlpatch::Route<'doc> {
+        let yamlpatch_components: Vec<yamlpatch::RouteComponent<'doc>> = self.components
+            .iter()
+            .map(|comp| match comp {
+                RouteComponent::Key(key) => yamlpatch::RouteComponent::Key(key),
+                RouteComponent::Index(idx) => yamlpatch::RouteComponent::Index(*idx),
+            })
+            .collect();
+        yamlpatch::Route::from(yamlpatch_components)
+    }
+}
```

Comment:
Can we do this with a `From<...>` impl instead? I think that'd be a bit easier to read :slightly_smiling_face: 

(Separately, I'd love to dedupe the three different route/query types we now have. But that can be a follow-on...)

Reply:
Fixed in https://github.com/zizmorcore/zizmor/pull/1001/commits/2c4a8c4ca76ac1b9fe17e44274438629e748468b.

(I like dedup, since the whole thing is getting out of hand. 😄)